fia_api/router.py

```diff
@@ -290,3 +294,52 @@ async def update_instrument_specification(
     """
     update_specification_for_instrument(instrument_name.upper(), specification)
     return specification
+
+
+@ROUTER.get("/extras", tags=["files"])
+async def get_extras_top_level_folders() -> list[str]:
+    """
+    Returns top level folders in the extras directory
+    \f
+    :return: List of folders
+    """
+    root_directory = Path(os.environ.get("EXTRAS_DIRECTORY", "/extras"))
+    safe_check_filepath(root_directory, root_directory)
+    return read_dir(root_directory)
+
+
+@ROUTER.get("/extras/{subdir}", tags=["files"])
+async def get_subfolder_files_list(subdir: str) -> list[Path]:
+    """
+    Returns a list of files within a sub_folder. Directs users to use the /extras endpoint if folder not found
+
+    :param subdir: The subdir to return the contents of
+    :return: List of files within a subdir
+    """
+    root_directory = Path(os.environ.get("EXTRAS_DIRECTORY", "/extras"))
+    subdir_path = root_directory / subdir
+    safe_check_filepath(subdir_path, root_directory)
+
+    return read_dir(subdir_path)
+
+
+@ROUTER.post("/extras/{instrument}/{filename}", tags=["files"])
+async def upload_file_to_instrument_folder(instrument: str, filename: str, file: UploadFile) -> str:
```

Comment:
Are there e2e tests for /extras/{}/{} the endpoint?

Reply:
Yes, https://github.com/fiaisis/FIA-API/blob/7c1d39051b49d0482fd2411c9ecc113a3ec5f7c6/test/e2e/test_file_upload.py#L117-L144 

But if all these tests were not easily identifiable, then maybe the test file and tests need a rename/restructure?